server/src/main/java/org/elasticsearch/cluster/metadata/MappingMetadata.java

```diff
@@ -73,10 +72,9 @@ public MappingMetadata(CompressedXContent mapping) {
     public MappingMetadata(String type, Map<String, Object> mapping) {
         this.type = type;
         try {
-            XContentBuilder mappingBuilder = XContentFactory.jsonBuilder().map(mapping);
-            this.source = new CompressedXContent(BytesReference.bytes(mappingBuilder));
-        }
-        catch (IOException e) {
+            this.source = new CompressedXContent(
```

Comment:
We get here a lot as part of some snapshot repo operations (when reading index metadata) and it's quite wasteful to first serialize the map to bytes into one buffer only to then compress that buffer to another buffer => just serialize and compress in one go